### PR TITLE
Run tests behind logged in pages

### LINKED ISSUE #141: Create a method of disabling login required during testing and development
⚠ Any method of disabling login creates a "backdoor". ⚠ 

- [ ] document in an ADR security considerations and how risks have been mitigated

### PR DESCRIPTION
**Pull request by SSPJ**
## 🗣 Description ##

Introduces a schema for running CI scans on pages which require login.

## 💭 Motivation and context ##

The way this works is by having a middleware class `registrar.tests.common.MockUserLogin` which checks every request received by the application. If the user is anonymous (as a scanning tool would be, upon first accessing the application), the middleware class creates a test user and logs the requester in.

This middleware class is physically copied into the list by the CI tool, meaning that it will not be present in the running code outside of the CI environment.

Closes #141 

Here's a screen shot taken by pa11y-ci, verifying a log in.

```
{
    "urls": [
        "http://app:8080/",
        "http://app:8080/health/",
        {
            "url": "http://app:8080/whoami/",
            "screenCapture": "screenshot.png"
        }
    ]
}
```

screenshot

### CODE REVIEW

File: src/registrar/tests/common.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def __call__(self, request):
+        if request.user.is_anonymous:
+            user = None
+            UserModel = get_user_model()
+            username = "Testy"
+            args = {
+                UserModel.USERNAME_FIELD: username,
+            }
+            try:
+                user = UserModel.objects.get(**args)
```
Comment (neilmb) [inline src/registrar/tests/common.py:18]: **suggestion (non-blocking):** I think that this `try/except` block could idiomatically be replaced with `UserModel.objects.get_or_create()`?

File: src/registrar/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
```
Comment (neilmb) [inline src/registrar/tests/common.py:1]: **question:** Is there a meaningful improvement in security if we keep this file outside of `src/` (which is what gets uploaded to CF) and using both a `cp` and the `perl` invocation to turn it on?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -35,6 +35,14 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Disable Login
```
Comment (neilmb) [inline .github/workflows/test.yaml:38]: **question:** Is there a recommended way to use this locally to run these same tests? If it is just "add this line to settings.py", could we document that in e.g. a (new) `docs/developer/testing.md`?

**Review by neilmb — APPROVED** [review]
LGTM. We should add that `perl` invocation to the OWASP tests whenever they or this PR lands.

File: src/registrar/tests/common.py (hunk 1 above)
Comment (SSPJ) [reply src/registrar/tests/common.py]: Good point. I've changed it.

File: .github/workflows/test.yaml (hunk 3 above)
Comment (SSPJ) [reply .github/workflows/test.yaml:38]: It is just "add this line to settings.py"... and I have updated the docs. :)

**Pull request merged**